File: Day17/Output/optimized.cs

using System;
using System.Numerics;

class Program
{
    static void Main(string[] args)
    {
        BigInteger iterations = 100_000_000;
        BigInteger param1 = 4;
        BigInteger param2 = 1;

        BigInteger result = 1.0;
        for (BigInteger i = 1; i <= iterations; i++)
        {
            BigInteger j = i * param1 - param2;
            result -= 1 / (BigInteger)j;

            j = i * param1 + param2;
            result += 1 / (BigInteger)j;
        }

        result *= 4;
        Console.WriteLine($"Result: {result:F12}");

        TimeSpan executionTime = TimeSpan.FromSeconds(DateTime.Now.Ticks / TimeSpan.TicksPerSecond - DateTime.UtcNow.Ticks / TimeSpan.TicksPerSecond);
        Console.WriteLine($"Execution Time: {executionTime.TotalSeconds:F6} seconds");
    }
}

File: Day17/Output/optimized.swift

import Foundation

func calculate(iterations: Int, param1: Double, param2: Double) -> Double {
    var result = 1.0
    for i in 1...iterations {
        let j1 = Double(i) * param1 - param2
        result -= 1 / j1
        let j2 = Double(i) * param1 + param2
        result += 1 / j2
    }
    return result
}

let startTime = Date()
let result = calculate(iterations: 100_000_000, param1: 4, param2: 1) * 4
let endTime = Date()

let executionTime = endTime.timeIntervalSince(startTime)

print("Result: \(result.truncatingRemainder(dividingBy: 1e-12))")
print("Execution Time: \(executionTime) seconds")
